### PR TITLE
Add NodeJS Year Service with OTLP

### LINKED ISSUE #44: Add node year service
This service returns a random year.

### PR DESCRIPTION
**Pull request by JamieDanielson**
Add NodeJS Year Service with OTLP
Adds support for environment variables

Side note: `//prettier-ignore`s added to keep imports from wrapping to multi-line with printWidth of 80.

Closes #44

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -148,13 +148,39 @@ def launch_dotnet_name_service(auto_init=True):
...
+    dirname: (optional) directory name in which to run `npm start` defaults to 'name'
+    flags: (optional) any additional flags to add to the command line
+    '''
+    
+    apikey = os.environ.get('HONEYCOMB_API_KEY')
+    dataset = os.getenv('HONEYCOMB_DATASET')
+    env = {
+        'SERVICE_NAME': name,
+        'HONEYCOMB_API_KEY': apikey,
+        'HONEYCOMB_DATASET': dataset,
```
Comment (vreynolds) [inline Tiltfile:166]: as a dogfood resident, may I request `HONEYCOMB_API` (for endpoint) as well?

File: node/docker-compose.yml [hunk 2]
```diff
@@ -1,12 +1,34 @@
 services:
+  collector:
```
Comment (vreynolds) [inline node/docker-compose.yml:2]: don't need the collector

File: node/year-service/main.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  resource: new Resource({
+    [ResourceAttributes.SERVICE_NAME]: `${SERVICE_NAME}`,
+  }),
+});
+
+const metadata = new grpc.Metadata();
+metadata.set('x-honeycomb-team', HONEYCOMB_API_KEY);
+metadata.set('x-honeycomb-dataset', HONEYCOMB_DATASET);
+
+provider.addSpanProcessor(
```
Comment (vreynolds) [inline node/year-service/main.js:53]: initialize tracing separately and include with `node -r`

**Review by vreynolds — CHANGES_REQUESTED** [review]
comment

**Comment by JamieDanielson** [comment]
* Split Open Telemetry to separate tracing file in Year service
* Added/updated environment variables, including OTLP_ENDPOINT
* Remove collector (not used in implementation)
* Added README for Node Services

File: Tiltfile [hunk 4]
```diff
@@ -148,13 +148,43 @@ def launch_dotnet_name_service(auto_init=True):
...
+    '''
+    Starts a single Node service.
+
+    Parameters:
+    name: used to display the name of the process in the tilt tab
+    dirname: (optional) directory name in which to run `npm start` defaults to 'name'
+    flags: (optional) any additional flags to add to the command line
+    '''
+    
+    apikey = os.getenv('HONEYCOMB_API_KEY')
```
Comment (vreynolds) [inline Tiltfile:161]: I don't think we need to do this mapping, the vars will be available to the process without `serve_env` (we only need to do that if we want to define variable in the Tilt file itself)

Comment (vreynolds) [reply Tiltfile:161]: Oh I think we would only keep the service name one, since that's defined through tilt

**Review by vreynolds — APPROVED** [review]
💥

File: Tiltfile (hunk 4 above)
Comment (JamieDanielson) [reply Tiltfile]: Makes sense - good call! I removed those extra variable declarations and confirmed `tilt up` still works based on env vars. Kept Service Name.

**Pull request merged**